concent_api/core/payments/sci_callback.py

```diff
@@ -0,0 +1,149 @@
+from contextlib import closing
+import socket
+
+from bitcoin import ecdsa_raw_verify
+from ethereum.utils import sha3
+from ethereum.transactions import UnsignedTransaction
+from ethereum.transactions import Transaction
+from golem_messages.exceptions import MessageError
+from rlp import encode
+
+from django.conf import settings
+
+from core.constants import SCI_CALLBACK_MAXIMUM_TIMEOUT
+from core.exceptions import SCICallbackFrameError
+from core.exceptions import SCICallbackPayloadError
+from core.exceptions import SCICallbackRequestIdError
+from core.exceptions import SCICallbackSignatureError
+from core.exceptions import SCICallbackTimeoutError
+
+from middleman_protocol.concent_golem_messages.message import SignedTransaction
+from middleman_protocol.concent_golem_messages.message import TransactionRejected
+from middleman_protocol.concent_golem_messages.message import TransactionSigningRequest
+from middleman_protocol.exceptions import MiddlemanProtocolError
+from middleman_protocol.message import ErrorFrame
+from middleman_protocol.message import GolemMessageFrame
+from middleman_protocol.stream import send_over_stream
+from middleman_protocol.stream import unescape_stream
+
+
+def sci_callback(transaction: Transaction) -> Transaction:
```

Comment:
this function is pretty big... Maybe we could have it split into smaller self-describing minifunctions (like in `blender_verification_order` case)?

Reply:
It has exactly 100 lines when you cut of beginning docstring, and there are no parts of it that are repeated. In `blender_verification_order` there was some part of repeated code, and mixed some low-level code with high-level code. I think this is not the case here, but if you insist I will do it. Maybe lets discuss tomorrow.

Reply:
@dybi Lets discuss tomorrow.

Reply:
for me 100 lines is a lot ;) if I cannot convince you, maybe Uncle Bob will? ;) 
https://vimeo.com/12643301

Reply:
:+1: done